Add missing dependency of database tests on backend

hramrach opened the pull request:
Missing make file dependency:

git clone https://github.com/go-gitea/gitea.git
make -C gitea  test-sqlite

go test  -c code.gitea.io/gitea/tests/integration -o integrations.sqlite.test -tags ' sqlite sqlite_unlock_notify'
sed -e 's|{{REPO_TEST_DIR}}||g' \
	-e 's|{{TEST_LOGGER}}|test,file|g' \
	-e 's|{{TEST_TYPE}}|integration|g' \
		tests/sqlite.ini.tmpl > tests/sqlite.ini
GITEA_ROOT="/home/hramrach/gitea" GITEA_CONF=tests/sqlite.ini ./integrations.sqlite.test
Could not find gitea binary at /home/hramrach/gitea/gitea

make: *** [Makefile:506: test-sqlite] Error 1

hramrach committed:
Makefile: Database tests need backend

fixes this error:

 make test-sqlite
go test  -c code.gitea.io/gitea/tests/integration -o integrations.sqlite.test -tags ' sqlite sqlite_unlock_notify'
sed -e 's|{{REPO_TEST_DIR}}||g' \
	-e 's|{{TEST_LOGGER}}|test,file|g' \
	-e 's|{{TEST_TYPE}}|integration|g' \
		tests/sqlite.ini.tmpl > tests/sqlite.ini
GITEA_ROOT="/home/hramrach/gitea" GITEA_CONF=tests/sqlite.ini ./integrations.sqlite.test
Could not find gitea binary at /home/hramrach/gitea/gitea

make: *** [Makefile:506: test-sqlite] Error 1

silverwind commented:
CI failures are unrelated and will be fixed with https://github.com/go-gitea/gitea/pull/36074.

hramrach committed:
Makefile: Enable CGO on non-arm

CGO is required for tests that use -race as well as some other build
options.

On arm64 the build fails when CGO is enabled because of some assembly
error.

hramrach committed:
pull-db-tests: Remove step building backend

No need to build it explicitly, db tests now depend on it.

silverwind commented:
Timeout after 50m seems odd as the db tests usually take only 30m, I've restarted the failing job, maybe it was a flake.

hramrach committed:
Try raising pgsql timeou

wxiaoguang commented:
> If the postgres test times out again, try raising this temporarily:
> 
> ```
> .github/workflows/pull-db-tests.yml:56:        timeout-minutes: 50
> ```

No, it shouldn't and don't do that. There must be a bug (eg: deadlock) if it timeouts.

wxiaoguang commented:
> It's still valuable if we see the output of a full test run that is not cancelled, otherwise you have no full picture of the errors to work with. Will of course be reverted before merge.

For testing-only purpose, you can try. But I guess it won't really help, because usually a test finishes in 20-30 minutes. So you already have the doubled time.
